tests/MongoDB.Bson.TestHelpers/Reflector.cs

```diff
@@ -22,7 +23,7 @@ public static class Reflector
     {
         public static object GetFieldValue(object obj, string name, BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Instance)
         {
-            var fieldInfo = obj.GetType().GetField(name, flags);
```

Comment:
Turns out GetField only looks at the current class, not at the base class. So we need a helper method that will go up the inheritance tree if necessary.